Issutracker : #iXXXXX#
CWS         : N/A
Author      : NAKATA Maho <maho@openoffice.org> (JCA)
Description : Get rid of -ldl for FREEBSD and MACOSX

Index: vcl/util/makefile.mk
===================================================================
RCS file: /cvs/gsl/vcl/util/makefile.mk,v
retrieving revision 1.65
diff -u -r1.65 makefile.mk
--- vcl/util/makefile.mk	13 Jan 2005 18:16:05 -0000	1.65
+++ vcl/util/makefile.mk	7 Apr 2005 01:29:29 -0000
@@ -252,10 +252,11 @@
 
 .IF "$(GUI)"=="UNX"
 
-.IF "$(OS)"=="MACOSX"
-SHL1STDLIBS += -ldl
+.IF "$(OS)"!="MACOSX" && "$(OS)"!="FREEBSD"
+SHL1STDLIBS+= -ldl
 .ENDIF
 
+
 .IF "$(GUIBASE)"=="aqua"
 SHL1STDLIBS += -framework Cocoa
 .ENDIF
@@ -383,7 +384,11 @@
 
 
 SHL4STDLIBS+=-l$(SHL2TARGET)
+.IF "$(OS)"=="FREEBSD" || "$(OS)"=="MACOSX"
+SHL4STDLIBS+=$(SHL3STDLIBS) -lX11
+.ELSE
 SHL4STDLIBS+=$(SHL3STDLIBS) -lX11 -ldl
+.ENDIF # "$(OS)"!="FREEBSD" || "$(OS)"!="MACOSX"
 .ENDIF # "$(ENABLE_GTK)" != ""
 
 # KDE plugin
@@ -396,7 +401,11 @@
 # libs for KDE plugin
 SHL5STDLIBS=$(KDE_LIBS)
 SHL5STDLIBS+=-l$(SHL2TARGET)
+.IF "$(OS)"=="FREEBSD" || "$(OS)"=="MACOSX"
+SHL5STDLIBS+=$(SHL3STDLIBS) -lX11
+.ELSE
 SHL5STDLIBS+=$(SHL3STDLIBS) -lX11 -ldl
+.ENDIF # "$(OS)"!="FREEBSD" || "$(OS)"!="MACOSX"
 .ENDIF # "$(ENABLE_KDE)" != ""
 
 .ENDIF # UNX
